freqtrade/freqtradebot.py

```diff
@@ -255,6 +267,8 @@ def _refresh_active_whitelist(self, trades: List[Trade] = []) -> List[str]:
         self.pairlists.refresh_pairlist()
         _whitelist = self.pairlists.whitelist
 
+        self.rpc.send_msg({'type': RPCMessageType.WHITELIST, 'data': _whitelist})
+
```

Comment:
This is what's causing tests in `test_freqtradebot.py` to fail (at least most of them).
The failures themselves are correct (we're now doing one additional "send_msg" call) - so they'll have to be adjusted.

Now to the problematic part - which should be worked on before the tests - as it may change the behavior of some tests):
This method is called (unconditionally) once every iteration (roughly every 5 seconds) - this means that the websocket will transmit (most likely the same) whitelist over and over (and over, ... ).

currently, that's not a problem - as we do need the uptodate pairlist - which may be identical - or different depending on the selected pairlists.
If we start emmitting the pairlist though - we'll need to add a filter here to only send it if it's different from the prior pairlist.

That should work just fine by comparing the new pairlist with a reference to `self.pairlists.whitelist` taken before `refresh_pairlist()`.

Reply:
Great idea, I will add this change :+1: